src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  checkTriangle(): boolean {
+    return this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+      || this.a >= this.b + this.c
+      || this.b >= this.a + this.c
+      || this.c >= this.a + this.b;
+  }
```

Comment:
Let's store these conditions in the array and check it with the Array.prototype.some() method